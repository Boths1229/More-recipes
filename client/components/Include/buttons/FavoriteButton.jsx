import React from 'react';
import PropTypes from 'prop-types';

const FavoriteButton = (props) => {
  const { favoriteRecipe, recipeId, favorites } = props;
  return (
    <div>
      <button className="btn btn-orange mr-2" type="button">
        <i
          className="fa fa-heart fa-2x favorite-btn"
          aria-hidden="true"
          onClick={() => favoriteRecipe(recipeId)}
        />
      </button>
      <span className="mr-2">
        {favorites}
      </span>
    </div>
  );
};

FavoriteButton.propTypes = {
  favoriteRecipe: PropTypes.func.isRequired,
  recipeId: PropTypes.number.isRequired,
  favorites: PropTypes.number.isRequired
};

export default FavoriteButton;
