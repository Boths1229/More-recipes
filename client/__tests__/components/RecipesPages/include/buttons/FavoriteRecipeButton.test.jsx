import React from 'react';
import { shallow } from 'enzyme';
import FavoriteButton from '../../../../../components/Include/buttons/FavoriteButton';

describe('Favorite button component', () => {
  it('matches snapshot', () => {
    const output = shallow(<FavoriteButton />);
    expect(output).toMatchSnapshot();
  });
  it('calls onClick', () => {
    const props = {
      favoriteRecipe: jest.fn()
    };
    const output = shallow(<FavoriteButton {...props} />);
    const button = output.find('i[aria-hidden="true"]');
    button.simulate('click');
    expect(props.favoriteRecipe).toHaveBeenCalled();
  });
});
