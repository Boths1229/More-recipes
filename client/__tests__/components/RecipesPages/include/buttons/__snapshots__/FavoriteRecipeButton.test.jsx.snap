// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite button component matches snapshot 1`] = `
<div>
  <button
    className="btn btn-orange mr-2 favorite"
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-heart fa-2x favorite-btn"
      onClick={[Function]}
    />
  </button>
  <span
    className="mr-2"
  />
</div>
`;
